stubs/CoreGenericIterators.phpstub

```diff
@@ -376,6 +376,11 @@ class CachingIterator extends IteratorIterator implements OuterIterator , ArrayA
      * @psalm-ignore-nullable-return
      */
     public function key() {}
+
+    /**
+     * @return array<array-key, TValue>
```

Comment:
That's unfortunate, I would have expected `TKey`, but it looks like it takes the value of the key and [casts it](https://3v4l.org/lkgM0) to `array-key`. I don't think we have a way to annotate that more specifically :slightly_frowning_face:

Reply:
That's indeed unfortunate and I couldn't find a way to type it better.